Sources/OpenAI/Private/StreamingSession.swift

```diff
@@ -72,29 +72,20 @@ extension StreamingSession {
                 return
             }
             guard let jsonData = jsonContent.data(using: .utf8) else {
-                onProcessingError?(self, StreamingError.unknownContent)
-                return
+                return onProcessingError?(self, StreamingError.unknownContent)
             }
             
-            var apiError: Error? = nil
             do {
                 let decoder = JSONDecoder()
                 let object = try decoder.decode(ResultType.self, from: jsonData)
                 onReceiveContent?(self, object)
             } catch {
-                apiError = error
-            }
-            
-            if let apiError = apiError {
-                do {
-                    let decoded = try JSONDecoder().decode(APIErrorResponse.self, from: jsonData)
+                if let decoded = try decoder.decode(APIErrorResponse.self, from: jsonData) {
```

Comment:
@ingvarus-bc This is a great catch, I made edits in GitHub's editor and didn't run it on my own machine to catch the compiler errors *ack!*. 

I actually think that a `try?` would be appropriate here because you don't really want to throw an error for a failed decode for the `APIErrorResponse`, it would be more beneficial to return the original error to the user where the decode failed for `ResultType` instead. That said, ideally the tests verify this shouldn't ever throw this error to the dev anyways.